src/main/java/frameworksdrivers/Gateways/StudentDbGateway.java

```diff
@@ -106,48 +106,52 @@ public void saveUser(CreateStudentDsModel student) {
-                statement.setString(1, student.getUsername());
-                statement.setString(2, student.getPassword());
-                statement.setString(3, course.getKey());
-                statement.setFloat(4, 0.0f);
-                statement.executeUpdate();}
+               saveUserNoCourses(student);
+            } else {
+                for (Map.Entry<String, Float> course : courses.entrySet()) {
+                    String SQL = "INSERT INTO " + DATABASE_NAME +
+                            " (StudentID, Password, CourseID, CourseGrade) VALUES (?, ?, ?, ?)";
+                    PreparedStatement statement = connection.prepareStatement(SQL);
+                    statement.setString(1, student.getUsername());
+                    statement.setString(2, student.getPassword());
+                    statement.setString(3, course.getKey());
+                    statement.setFloat(4, 0.0f);
+                    statement.executeUpdate();}
             }
         } catch (SQLException e) {
             System.out.println("Error with database!");
             e.printStackTrace();
         }
     }
 
-    /**This method saves a user when they first open a program, and they have no courses created
-     * @param student this is the request model containing the student password and username*/
+    /**
+     * This method saves a user when they first open a program, and they have no courses created
+     *
+     * @param student this is the request model containing the student password and username
+     */
     private void saveUserNoCourses(CreateStudentDsModel student){
-        try {String SQL = "INSERT INTO " + DATABASE_NAME +
+        try {
+            String SQL = "INSERT INTO " + DATABASE_NAME +
                 " (StudentID, Password, CourseID, CourseGrade) VALUES (?, ?, ?, ?)";
-        PreparedStatement statement = connection.prepareStatement(SQL);
-        statement.setString(1, student.getUsername());
-        statement.setString(2, student.getPassword());
-        statement.setString(3, "");
-        statement.setFloat(4, 0.0f);
-        statement.executeUpdate();
-    }
-        catch(SQLException e){
+            PreparedStatement statement = connection.prepareStatement(SQL);
+            statement.setString(1, student.getUsername());
+            statement.setString(2, student.getPassword());
+            statement.setString(3, "");
+            statement.setFloat(4, 0.0f);
+            statement.executeUpdate();
+        } catch(SQLException e) {
             System.out.println("Error with database!");
-            e.printStackTrace();}
+            e.printStackTrace();
         }
+    }
 
 
-        /**
-         * Saves the GPA of a student with a username and courseID. The courseID specifies what course the grade is from,
-         * the username specifies what student got this grade.
-         *
-         * @param requestModel with the StudentID, CourseID, and grade.
-         */
```

Comment:
Fixed some of the indentation here to be consistent with the function method header